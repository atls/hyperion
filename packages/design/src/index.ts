export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
