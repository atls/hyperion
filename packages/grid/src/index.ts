export * from './Column'
export * from './Grid'
