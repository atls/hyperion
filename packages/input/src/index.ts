export * from './Textarea'
export * from './Input'
export * from './Checkbox'
