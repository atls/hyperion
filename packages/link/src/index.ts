export * from './Link'
export * from './NextLink'
export * from './RouteLink'
