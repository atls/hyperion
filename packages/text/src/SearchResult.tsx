import React  from 'react'

import styled from '@emotion/styled'

const Matched = styled.span({
  background: '#396aab',
})

export const SearchResult = ({ children, match }) => {
  if (!match) {
    return children
  }

  const start = children.toLowerCase().indexOf(match)

  if (start === -1) {
    return children
  }

  const { length } = match

  return [
    <span
      key='start'
      dangerouslySetInnerHTML={{
        __html: children.substr(0, start).replace(/ /g, '&nbsp;'),
      }}
    />,
    <Matched
      key='value'
      dangerouslySetInnerHTML={{
        __html: children.substr(start, length).replace(/ /g, '&nbsp;'),
      }}
    />,
    <span
      key='end'
      dangerouslySetInnerHTML={{
        __html: children.substr(start + length).replace(/ /g, '&nbsp;'),
      }}
    />,
  ]
}
