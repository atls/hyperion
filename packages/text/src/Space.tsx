import React from 'react'

export const Space = ({ count = 1 }) => (
  <span style={{ display: 'inline-flex' }}>{'\u00A0'.repeat(count)}</span>
)
