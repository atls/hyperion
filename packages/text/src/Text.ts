import { color, space, system, typography } from 'styled-system'

import styled from '@emotion/styled'

export const Text = styled.span(
  system({
    wordBreak: true,
    whiteSpace: true,
    textTransform: true,
    textOverflow: true,
    cursor: true,
  }),
  color,
  space,
  typography
)

Text.defaultProps = {
  fontFamily: 'primary',
  fontWeight: 'normal',
  fontSize: 'normal',
}
