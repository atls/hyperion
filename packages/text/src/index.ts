export * from './Text'
export * from './Space'
export * from './SearchResult'
