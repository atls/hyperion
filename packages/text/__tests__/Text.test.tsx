import React    from 'react'
import renderer from 'react-test-renderer'

import { Text } from '../src'

it('should render Text component correctly', () => {
  const tree = renderer
    .create(
      <Text color='red' fontFamily='Courier' fontSize='2vh'>
        This is test
      </Text>,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
