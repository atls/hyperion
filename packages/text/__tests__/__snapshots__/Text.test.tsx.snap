// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Text component correctly 1`] = `
.emotion-0 {
  color: red;
  font-family: Courier;
  font-size: 2vh;
  font-weight: normal;
}

<span
  className="emotion-0"
  color="red"
  fontFamily="Courier"
  fontSize="2vh"
  fontWeight="normal"
>
  This is test
</span>
`;
